sw/device/lib/dif/BUILD

```diff
@@ -0,0 +1,657 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+  name = "clkmgr",
+  srcs = [
+      "dif_clkmgr.c",
+  ],
+  hdrs = ["dif_clkmgr.h"],
+  deps = [
+      "//sw/device/lib/base",
+      "//sw/device/lib/base:mmio",
```

Comment:
You should not supply `base:mmio` here.  The target `base` supplies the mmio _interface_ via `mmio.h`.  Some later rule (ie: cc_{binary,test}) should supply the _implementation_ by linking with `base:mmio` or `base/testing`.

The `clkmgr_unittest` rule does just that: it depends on `base/testing` which supplies the mock_mmio implementation.

Reply:
I thought as soon as we were building without the MOCK_MMIO flag we should just depend on the implementation, but yes we can do that for this and the other dif libraries.